ckanext/cfpb_extrafields/fanstatic/post_related_gist.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+ckan.module('post_related_gist', function ($, _) {
+    return {
+        initialize: function () {
+            $.proxyAll(this, /_on/);
+            this.el.on('click', this._onClick);
+        },
+
+        _onClick: function(event) {
+            var editor = ace.edit("editor_div");
+            this.options.gistdesc = $('#gist-description').val();
+            if (this.options.gistdesc == ''){
+                var outhtml = '<h3><font color="red">Fill in description field.</font></h3>';
+                $('#ace_output').html(outhtml);
+                $('#gist-description').focus();
+                return;
+            }
+            var textgist = editor.getSession().getValue();
+            $('#ace_output').html('<div id="loader"><img src="/loader.gif" alt="loading..."></div>');
+            var apiurl   = 'https://github.cfpb.gov/api/v3/gists';
+            
+            this._postGIST(apiurl,textgist, function(json) {
+                if (!("html_url" in json)){
+                    $('#ace_output').html("<h2>GitHub post failed</h2><br>",json);
+                }else {
+                    this.options.gistlink = json.html_url;
+                    var outhtml = '<h3>Posted a <a href="'+json.html_url+'">Gist</a></h3>';
+                    $('#ace_output').html(outhtml);
+                    var option = $('<option></option>').attr("value", this.options.gistlink).text(this.options.gistdesc);
+                    $('#gistselect').append(option); 
+                    
+                    if (!this._snippetReceived) {
+                        this.sandbox.client.getTemplate(this.options.html,
+                                                        this.options,
+                                                        null);
+                        this._snippetReceived = true;
+                    }
+                } 
+            });
+        },
+
+        _postGIST: function(apiurl, basecontent, callback) {
+            var mythis=this; // code review?
```

Comment:
Hey folks. By setting mythis = this at the top of the function you ensure that you're binding the correct contenxt for your $.ajax call further down the page. If you were to pass "this" into the complete function, you'd end up passing a "this" value bound to that call site, and NOT the top level _postGIST "this" value that you intended, so probably a good idea to assign it as you did.

Reading this again, it looks like you could probably use a "this" when assigning the `.options.gistdesc` value on line 45 without issue, but as soon as you try to call it in a nested function the scope changes unless you .bind() to another scope. Either way I'd stick with it once you've assigned "mythis" if you need to re-use it again so you can keep your head straight reading the function in the future.

Hope that makes sense.  [Here's a good clarification doc if you're interested - that and chapter two are good things to know regarding the peculiarities of "this" in JS.](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20&%20object%20prototypes/ch1.md)